src/tools/simeng/gdb/GDBStub.cc

```diff
@@ -0,0 +1,375 @@
+#include "simeng/GDBStub.hh"
+
+namespace simeng {
+
+GDBStub::GDBStub(simeng::Core& core, simeng::MemoryInterface& dataMemory)
+    : core_(core), dataMemory_(dataMemory) {}
+
+int GDBStub::run() {
+  // verbose_ = true; // uncomment this if you'd like the GDBStub to run
+  // verbosely
+
+  int connection = openSocket(2424);  // change this if the port is blocked
+
+  std::cout << "[SimEng:GDBStub] Connection to GDB client established, "
+               "debugging in progress"
+            << std::endl
+            << std::endl;
+
+  char buffer[10000];
```

Comment:
Why has 10,000 been chosen? Is there a need for a buffer to be ~10KiB?

Reply:
The only reason I remember was just that the buffer would be long enough.  10,000 is excessive here.  The longest incoming message I see in current normal usage is 156 characters.  I expect this to rise when run/start is supported, but I suggest it could be 300 or so for now